src/Amalgam/interpreter/InterpreterOpcodesBase.cpp

```diff
@@ -705,8 +706,14 @@ EvaluableNodeReference Interpreter::InterpretNode_ENT_DECLARE(EvaluableNode *en,
 			{
 				if(need_to_interpret && cn != nullptr && !cn->GetIsIdempotent())
 				{
+				#ifdef MULTITHREAD_SUPPORT
+					Concurrency::WriteLock write_lock(*callStackMutex, std::defer_lock);
+					if(callStackMutex != nullptr && GetCallStackDepth() < callStackUniqueAccessStartingDepth)
```

Comment:
If callStackMutex can be null, you deref in line 710 but only check for null after in line 711.